File: Pod/Classes/PUROutput.m
#import "PUROutput.h"
#import "PURLogger.h"

@implementation PUROutput

- (instancetype)initWithLogger:(PURLogger *)logger tagPattern:(NSString *)tagPattern
{
    self = [super init];
    if (self) {
        _identifier = [NSUUID UUID].UUIDString;
        _tagPattern = tagPattern;
        _logger = logger;
    }
    return self;
}

- (PURLogStore *)logStore
{
    return self.logger.logStore;
}

- (void)configure:(NSDictionary<NSString *, id> *)settings
{

}

- (void)start
{

}

- (void)resume
{

}

- (void)suspend
{

}

- (void)emitLog:(PURLog *)log
{

}

@end

File: Pod/Classes/PUROutput.swift
//
//  PUROutput.swift
//  Pods
//
//  Created by admin on 7/27/17.
//
//

import Foundation

class PUROutput {
    private(set) var identifier: String = ""
    private(set) var tagPattern: String = ""
    private(set) var logger: PURLogger?
    private(set) var logStore: PURLogStore?
    
    init(logger: PURLogger,
         tagPattern: String) {
        self.identifier = UUID().uuidString
        self.tagPattern = tagPattern
        self.logger = logger
    }
    
    func getLogStore() -> PURLogStore? {
        if let logStore = logStore {
            return logStore
        }
        
        return nil
    }
    
    func configure(_ settings: [String: Any]) {
    }
    
    func start() {
    }
    
    func resume() {
    }
    
    func suspend() {
    }
    
    func emitLog(_ log: PURLog) {
    }
}
